File: medium/sort_array.go
package medium

func SortArray(nums []int) []int {
	if len(nums) == 0 {
		return nums
	}
	freq := make(map[int]int)
	result := make([]int, len(nums))
	index, max, min := 0, nums[0], nums[0]
	for _, num := range nums {
		freq[num]++
		if max < num {
			max = num
		}
		if min > num {
			min = num
		}
	}
	for i := min; i <= max; i++ {
		count := freq[i]
		for count > 0 {
			result[index] = i
			count--
			index++
		}
	}
	return result
}

File: medium/sort_array.py
from collections import Counter


def sort_array(nums: list[int]) -> list[int]:
    if len(nums) == 0:
        return []
    freq = Counter(nums)
    result = [0] * len(nums)
    mini, maxi, index = min(nums), max(nums), 0
    i = mini
    while i <= maxi:
        count = freq.get(i) or 0
        while count > 0:
            count -= 1
            result[index] = i
            index += 1
        i += 1
    return result
